### PR TITLE
Dr2 1547 generate new ocfl structure

### PR DESCRIPTION
**Pull request by techncl**

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/OcflService.scala [hunk 1]
```diff
@@ -53,13 +52,17 @@ class OcflService(ocflRepository: OcflRepository) {
...
+              potentialFile match {
+                case Some(ocflFileObject) =>
+                  val checksumUnchanged =
+                    Some(ocflFileObject.getFixity.get(DigestAlgorithm.sha256)).contains(obj.checksum)
+                  if (checksumUnchanged) objectMap else objectMap + ("changedObjects" -> (obj :: changedObjects))
+                case None => objectMap + ("missingObjects" -> (obj :: missedObjects)) // IO exists but file doesn't
+              }
 
-            case Failure(objectNotFound: NotFoundException) => objectMap + ("missingObjects" -> (obj :: missedObjects))
+            case Failure(objectNotFound: NotFoundException) =>
```
Comment (MancunianSam) [inline src/main/scala/uk/gov/nationalarchives/OcflService.scala:64]: The `objectNotFound` isn't used. Is it best to replace it with `_` ?

File: src/main/scala/uk/gov/nationalarchives/Processor.scala [hunk 2]
```diff
@@ -29,52 +30,139 @@ class Processor(
...
         parentRef <- IO.fromOption(entity.parent)(new Exception("Cannot get IO reference from CO"))
-      } yield bitstreamInfoPerCo.toList.map(bitStreamInfo =>
-        FileObject(parentRef, bitStreamInfo.name, bitStreamInfo.fixity.value, bitStreamInfo.url)
-      )
+        urlsOfRepresentations <- entityClient.getUrlsToIoRepresentations(parentRef, None)
+        coRepTypes <- urlsOfRepresentations.map(getRepresentationTypeOfCo(parentRef, _, entity.ref)).flatSequence
+        _ <- IO.raiseWhen(coRepTypes.length > 1) {
+          new Exception(s"${entity.ref} belongs to more than 1 representation type: ${coRepTypes.mkString(", ")}")
+        }
+        representationTypeGroup = coRepTypes.head
```
Comment (MancunianSam) [inline src/main/scala/uk/gov/nationalarchives/Processor.scala:117]: If you do `headOption` here then you don't have to wrap it in `Some` further down.

File: src/main/scala/uk/gov/nationalarchives/Processor.scala [hunk 3]
```diff
@@ -29,52 +30,139 @@ class Processor(
...
+
+  private def createMetadataFileName(entityType: String) = s"${entityType}_Metadata.xml"
+
   private def toDisasterRecoveryObject(message: Message): IO[List[DisasterRecoveryObject]] = message match {
     case InformationObjectMessage(ref, _) =>
       for {
         entity <- fromType[IO](EntityClient.InformationObject.entityTypeShort, ref, None, None, deleted = false)
-        metadata <- entityClient.metadataForEntity(entity).map { metadata =>
-          createMetadataObject(entity, metadata)
+        metadataFileName = createMetadataFileName(entity.entityType.get.entityTypeShort)
```
Comment (MancunianSam) [inline src/main/scala/uk/gov/nationalarchives/Processor.scala:94]: As you've created the entity from the type, can you just pass in `InformationObject.entityTypeShort` to `createMetadataFileName`? It gets rid of the `.get`

File: src/test/scala/uk/gov/nationalarchives/MainTest.scala [hunk 4]
```diff
@@ -282,4 +486,99 @@ class MainTest extends AnyFlatSpec with MockitoSugar with EitherValues {
...
+    metadataContent must equal(
+      trim(<AllMetadata>
+        <DifferentMetadata></DifferentMetadata>
+      </AllMetadata>)
+    )
+  }
+
+  "runDisasterRecovery" should "throw an error if the OCFL repository returns an unexpected error" in {
+    val id = UUID.randomUUID()
+    val repoDir = Files.createTempDirectory("repo")
```
Comment (MancunianSam) [inline src/test/scala/uk/gov/nationalarchives/MainTest.scala:569]: This variable and `expectedMetadataFileDestinationFilePath` aren't being used in this test.

File: src/main/scala/uk/gov/nationalarchives/OcflService.scala [hunk 5]
```diff
@@ -53,13 +52,17 @@ class OcflService(ocflRepository: OcflRepository) {
...
-                  .map(_.getFixity.get(DigestAlgorithm.sha256))
-                  .contains(obj.checksum)
-              if (checksumUnchanged) objectMap else objectMap + ("changedObjects" -> (obj :: changedObjects))
+              val potentialFile = Option(ocflObject.getFile(obj.destinationFilePath))
+              potentialFile match {
+                case Some(ocflFileObject) =>
+                  val checksumUnchanged =
+                    Some(ocflFileObject.getFixity.get(DigestAlgorithm.sha256)).contains(obj.checksum)
+                  if (checksumUnchanged) objectMap else objectMap + ("changedObjects" -> (obj :: changedObjects))
+                case None => objectMap + ("missingObjects" -> (obj :: missedObjects)) // IO exists but file doesn't
```
[the remark this answers is not in the document]
Comment (techncl) [reply src/main/scala/uk/gov/nationalarchives/OcflService.scala:61]: 😄 sure

**Review by MancunianSam — APPROVED** [review]

**Pull request merged**